### ISSUE TITLE
Externalizing node properties silently introduces Defer semantics upstream

### ISSUE THREAD

**Issue by glopesdev** [bonsai-rx/bonsai#851]
Certain operators such as `Timer` or `Range` require their properties to be set at observable creation time (i.e. when the `Generate` or `Process` method is called) in order for them to be picked up during initialization.

To ensure that externalized properties have a chance to be initialized before their target observable is constructed, a `Defer` operator is introduced to delay the creation of the observable until after the properties have been initialized.

https://github.com/bonsai-rx/bonsai/blob/4b52d8656aecac34eca77976a960f32045e3e752/Bonsai.Core/Expressions/ExpressionBuilder.cs#L1113

However, the straight use of the `Defer` operator means that every subsequent subscription will recreate the observable from scratch, which is highly surprising and unexpected behavior and violates the intended semantics of workflows. Observable sequences should be constructed only once and then resubscribed to as necessary. If re-construction on each subscription is indeed required, the `Defer` group operator node provides explicit support for this.

The implicit introduction of the `Defer` operator for externalized properties is therefore a major issue which should be addressed as soon as possible. The intended pattern is likely more similar to lazy initialization, where the constructor is indeed deferred, but only the first time the expression is evaluated. For instance, replacing the use of `Defer` with the following `Lazy` operator would suffice to fix this:

```c#
static IObservable<TResult> Lazy<TResult>(Func<IObservable<TResult>> observableFactory)
{
    var source = new Lazy<IObservable<TResult>>(observableFactory);
    return Observable.Defer(() => source.Value);
}
```

Regression tests should be put in place to monitor the correct implementation of these specific semantics going forward, and it is likely that existing workflows might suffer from the bug-fix itself. In these cases, the "workaround" to restore the correct and intended semantics is to wrap affected workflows inside a `Defer` group.
